modules/http-api/src/main/java/org/mule/runtime/http/api/client/HttpClient.java

```diff
@@ -166,4 +171,34 @@ default CompletableFuture<WebSocket> openWebSocket(HttpRequest request,
                                                      WebSocketCallback callback) {
     throw new UnsupportedOperationException("WebSockets are only supported in Enterprise Edition");
   }
+
+  /**
+   * Creates a consumer of Server-sent events. The resulting {@link SseSource} is not connected automatically.
+   * <p>
+   * This API is EXPERIMENTAL. Do not use it until it is stable.
+   *
+   * @param url         the URL of the server.
+   * @param retryConfig configuration for the retry mechanism.
+   * @return a non-connected instance of {@link SseSource}.
+   * @since 4.10.0
+   */
+  @Experimental
+  default SseSource sseSource(String url, SseRetryConfig retryConfig) {
```

Comment:
i see terms like `consumer`, `source` and `listener` used interchangeably. Please unify the terminology so the api is more understandable.